src/documents/tutorial-developerIntroduction/DeveloperIntroductionToInfusionFramework-EventsAndInversionOfControl.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Developer Introduction to the Infusion Framework - Events and Inversion of Control
+layout: default
+category: Tutorials
+---
+
+All Infusion components support highly flexible event-driven programming. All components have basic [lifecycle events](/infusion/development/ComponentLifecycle.md) such as creation and destruction, and others may be [declared](/infusion/development/InfusionEventSystem.md#declaring-an-event-on-a-component), [fired](/infusion/development/InfusionEventSystem.md#using-events-and-listeners-procedurally) and [listened](/infusion/development/InfusionEventSystem.md#registering-a-listener-to-an-event) for by the originating component or another component.
+
+Infusion also makes extensive use of the programming concept of [Inversion of Control](/infusion/development/FrameworkConcepts.md#ioc) (IoC), a technique for organizing component dependencies and references in a distributed, flexible manner. IoC is used in many ways throughout Infusion, but an important initial concept to grasp is the use of [IoC References](/infusion/development/IoCReferences.md) when configuring components, and the concept of `{that}`.
+
+IoC references allow us to refer to other parts of a component object (or another component entirely) in a declarative, context-specific manner, with `{that}` standing in for the current component configuration.
+
+We can use a listener definition, the `onCreate` lifecycle event and IoC `{that}` to have a component say hello when it's ready, rather than needing to be manually called:
+
+<div class="infusion-docs-note">You can check out the [Live Example](http://codepen.io/waharnum/pen/LxbPQZ?editors=1111) of the code below on [CodePen](http://codepen.io/)</div>
+
+```
+fluid.defaults("fluid.helloWorld", {
+    gradeNames: ["fluid.component"],
+    listeners: {
+        // Configures the component to call its own 'sayHello' function
+        // when it's created
+        //
+        // On the left side, "onCreate.sayHello":
+        // 1) Configures the listener to listen for
+        // the onCreate lifecycle event
+        // 2) Adds an a namespace of 'sayHello' to
+        // allow multiple listeners to be attached to the
+        // same event without collision
+        //
+        // On the right side, this configures the listener
+        // to use the sayHello invoker 
+        "onCreate.sayHello": "{that}.sayHello"
+    },
+    invokers: {
+        sayHello: {
+            "this": "console",
```

Comment:
Experimenting with using `fluid.log`, it seems necessary to do one of two things to use it as a substitute for `console.log` in this context...

1) Use the two-argument form: `[fluid.logLevel.IMPORTANT, "Hello, World!!"]`.
2) Call `fluid.setLogging(fluid.logLevel.INFO)` at some point before the call.

As much as I'd like to keep things simple and short as possible, this may simply require more expansion to be a really adequate explanation of what invokers "are" and the purpose they serve; I think it is probably important to comprehend both the Infunion-ist "that" (function references should be globally resolvable) from the reality of much of the rest of the Javascript world.

At the moment my feeling is that it is important to highlight three uses (as described at http://docs.fluidproject.org/infusion/development/Invokers.html):

1) Referring to free functions 
2) Referring to IoC resolved functions
3) "this"-ism

I'll continue to think on this one and revise accordingly.